packages/order-watcher/src/order_watcher/order_watcher_websocket.ts

```diff
@@ -0,0 +1,184 @@
+        this.httpServer.close();
+    }
+
+    private _messageCallback(connection: WebSocket.connection, message: any): void {
+        assert.doesConformToSchema('message', message, webSocketUtf8MessageSchema);
+        const requestData: WebSocketRequestData = JSON.parse(message.utf8Data);
+        const responseData = this._routeRequest(requestData);
+        logUtils.log(`${new Date()} [Server] OrderWatcher output: ${JSON.stringify(responseData)}`);
+        connection.sendUTF(JSON.stringify(responseData));
+    }
+
+    private _closeCallback(connection: WebSocket.connection): void {
+        this._connectionStore.delete(connection);
+        logUtils.log(`${new Date()} [Server] Client ${connection.remoteAddress} disconnected.`);
+    }
+
+    private _routeRequest(requestData: WebSocketRequestData): WebSocketResponseData {
+        const responseData: WebSocketResponseData = {
+            action: requestData.action,
+            success: 0,
+            result: undefined,
+        };
+
+        try {
+            logUtils.log(`${new Date()} [Server] Request received: ${requestData.action}`);
+            switch (requestData.action) {
+                case 'addOrderAsync': {
+                    const signedOrder: SignedOrder = this._parseSignedOrder(requestData);
+                    // tslint:disable-next-line:no-floating-promises
+                    this._orderWatcher.addOrderAsync(signedOrder); // Ok to fireNforget
+                    break;
+                }
+                case 'removeOrder': {
+                    this._orderWatcher.removeOrder(requestData.params.orderHash);
+                    break;
+                }
+                case 'getStats': {
+                    responseData.result = this._orderWatcher.getStats();
+                    break;
+                }
+                default:
+                    throw new Error(`[Server] Invalid request action: ${requestData.action}`);
+            }
+            responseData.success = 1;
+        } catch (err) {
+            responseData.result = { error: err.toString() };
+        }
+        return responseData;
+    }
+
+    /**
+     * Broadcasts OrderState changes to ALL connected clients. At the moment,
+     * we do not support clients subscribing to only a subset of orders. As such,
+     * Client B will be notified of changes to an order that Client A added.
+     */
+    private readonly _broadcastCallback: OnOrderStateChangeCallback = (err, orderState) => {
+        this._connectionStore.forEach((connection: WebSocket.connection) => {
+            const responseData: WebSocketResponseData = {
+                action: OrderWatcherAction.orderWatcherUpdate,
+                success: 1,
+                result: orderState || err,
+            };
+            connection.sendUTF(JSON.stringify(responseData));
+        });
+        // tslint:disable-next-line:semicolon
+    }; // tslint thinks this is a class method, It's actally a property that holds a function.
```

Comment:
Hm when I make it a class method, the tslint rule prefer-function-to-method calls me out because these functions don't use `this`. That made sense to me because the functions just wrap a bunch of code instead of mutating aspects of the object (as methods typically do). 

But I don't feel strongly. Should I overwrite the `prefer-function-over-method` lint rule and change these to class methods?